### PR TITLE
Require PathUtils instead of using the global one

### PR DESCRIPTION
**Pull request by ficristo**
Actually I decided to try again: similarly to https://github.com/adobe/brackets/pull/11734 but this time using the submodule.

### CODE REVIEW

File: src/brackets.js [hunk 1]
```diff
@@ -115,6 +114,18 @@ define(function (require, exports, module) {
         }
     });
 
+    // DEPRECATED: In future we want to remove the global PathUtils, but for now we
+    // expose our required PathUtils globally so as to avoid breaking extensions in the
+    // interim.
+    var PathUtils = require("thirdparty/path-utils/path-utils");
+
+    Object.defineProperty(window, "PathUtils", {
```
Comment (MiguelCastillo) [inline src/brackets.js]: I like this. Thank you!

**Comment by MiguelCastillo** [comment]
These changes look good.  When we add breaking changes, we tend to notify extension authors.  @MarcelGerber, do you think you can help here please?

**Comment by MarcelGerber** [comment]
There are quite a few extensions that still use the global object:
* aem-sightly-brackets-extension
* brackets-latex
* cf10.cfbrackets.org
* cf11.cfbrackets.org
* cf7.cfbrackets.org
* cf8.cfbrackets.org
* cf9.cfbrackets.org
* luceebrackets.org
* nicolo-ribaudo.brackets-zip
* obd3.cfbrackets.org
* railo2.cfbrackets.org
* railo3.cfbrackets.org
* railo4.cfbrackets.org


But, as I see this, this is not a breaking change, as the global object is still accessible, it just outputs an deprecation warning when used. (Am I right?)

**Comment by ficristo** [comment]
Yes, this PR come with a deprecation warning, it shouldn't break extensions.

**Comment by MarcelGerber** [comment]
Same as the Mustache one, this one looks fine. Just that one thing that you could remove the Gruntfile line goes for this, too.

**Comment by ficristo** [comment]
@MarcelGerber  done.

**Comment by MarcelGerber** [comment]
Ah, your Mustache changes don't go well together with this one.

**Comment by ficristo** [comment]
Fixed (I hope).

**Comment by MarcelGerber** [comment]
Looks good now. Thank you!

**Pull request merged**